### PR TITLE
add pod annotation and make secret object optional

### PR DESCRIPTION
**Pull request by rabunkosar-dd**
- Adding ability to annotate pods
- Making secret creation optional based on the secret values
- Adding option to add bearer token for prometheus
- Updated values.yaml accordingly

### CODE REVIEW

File: charts/opencost/templates/deployment.yaml [hunk 1]
```diff
@@ -74,6 +78,13 @@ spec:
                   name: {{ include "opencost.fullname" . }}
                   key: DB_BASIC_AUTH_PW
             {{- end }}
+            {{- if .Values.opencost.prometheus.bearer_token }}
+            - name: DB_BEARER_TOKEN
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "opencost.fullname" . }}
+                  key: DB_BEARER_TOKEN
+            {{- end }}
```
Comment (Pokom) [inline charts/opencost/templates/deployment.yaml:87]: Only think I'd suggest here is to ensure only the `password` or the `bearer_token` can be set, and not both at once. Not a hard stop to getting this out, but would be nice to have.

Comment (rabunkosar-dd) [reply charts/opencost/templates/deployment.yaml:87]: good suggestion, I'll add that check

Comment (Pokom) [reply charts/opencost/templates/deployment.yaml:87]: Sounds good. Just re-request a review once you're ready and I can take another look.

**Review by Pokom — APPROVED** [review]
:+1: I'm assuming you've ran this @rabunkosar-dd?

**Comment by rabunkosar-dd** [comment]
> 👍 I'm assuming you've ran this @rabunkosar-dd?

Thanks for pointing that out. I did a last minute run, looks like the pod annotation wasn't properly indented, which I didn't test properly. Fixed it now. Others are fine (username/password/bearer_token)

**Review by Pokom — APPROVED** [review]
:+1:

**Comment by Pokom** [comment]
Going to revert because the publish stage failed. Root cause is we forgot to update the chart version.

**Pull request merged**